src/Console/LogVerbosity.php

```diff
@@ -9,9 +9,51 @@
 
 namespace Infection\Console;
 
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
 final class LogVerbosity
 {
-    const DEBUG = 1;
-    const NORMAL = 2;
-    const NONE = 3;
+    const DEBUG = 'all';
+    const NORMAL = 'default';
+    const NONE = 'none';
+
+    /**
+     * @deprecated
+     */
+    const DEBUG_INTEGER = 1;
+
+    /**
+     * @deprecated
+     */
+    const NORMAL_INTEGER = 2;
+
+    /**
+     * @deprecated
+     */
+    const NONE_INTEGER = 3;
+
+    const ALLOWED_OPTIONS = [
+        self::DEBUG_INTEGER => self::DEBUG,
+        self::NORMAL_INTEGER => self::NORMAL,
+        self::NONE_INTEGER => self::NONE,
+    ];
+
+    public static function convertVerbosityLevel(InputInterface $input, SymfonyStyle $io)
+    {
+        $verbosityLevel = $input->getOption('log-verbosity');
+        if (in_array($verbosityLevel, self::ALLOWED_OPTIONS)) {
+            return;
+        }
+
+        if (array_key_exists((int) $verbosityLevel, self::ALLOWED_OPTIONS)) {
+            $input->setOption('log-verbosity', self::ALLOWED_OPTIONS[$verbosityLevel]);
+            $io->note('Numeric versions of log-verbosity have been deprecated, please use, ' . self::ALLOWED_OPTIONS[$verbosityLevel] . ' to keep the same result');
+
+            return;
+        }
+
+        $io->note('Running infection with an unknown log-verbosity option, falling back to \'default\' option');
```

Comment:
Since the logging of the results does not change the output of msi/ what mutators are used etc we allow the process to continue, but we do give the end users a notice if they use an unknown value.